src/net/spdy/spdy_session.cc

```diff
@@ -2172,11 +2184,30 @@ void SpdySession::EnqueueResetStreamFrame(spdy::SpdyStreamId stream_id,
   });
 
   DCHECK(buffered_spdy_framer_.get());
+  int padding_len = base::RandInt(48, 72) - spdy::kDataFrameMinimumSize;
   std::unique_ptr<spdy::SpdySerializedFrame> rst_frame(
-      buffered_spdy_framer_->CreateRstStream(stream_id, error_code));
+      buffered_spdy_framer_->CreateRstStream(stream_id, error_code,
+                                             padding_len));
+
+  auto buffer = std::make_unique<SpdyBuffer>(std::move(rst_frame));
+
+  DecreaseSendWindowSize(padding_len);
+  buffer->AddConsumeCallback(base::BindRepeating(
+      &SpdySession::OnWriteBufferConsumed, weak_factory_.GetWeakPtr(),
+      static_cast<size_t>(padding_len)));
 
-  EnqueueSessionWrite(priority, spdy::SpdyFrameType::RST_STREAM,
-                      std::move(rst_frame));
+  auto it = active_streams_.find(stream_id);
+  if (it != active_streams_.end()) {
+    it->second->DecreaseSendWindowSize(padding_len);
```

Comment:
This part is actually unnecessary because you're immediately tearing down the stream.